src/mesido/workflows/utils/adapt_profiles.py

```diff
@@ -76,97 +75,163 @@ def adapt_hourly_year_profile_to_day_averaged_with_hourly_peak_day(problem, prob
 
     for ensemble_member in range(problem.ensemble_size):
         parameters = problem.parameters(ensemble_member)
-        total_demand = None
+        
+        total_heat_demand = None
         heat_demand_nominal = dict()
-
-        for demand in demands:
+        # Assemble all demands together to get the peaks.
+        for demand in heat_demands:
             try:
                 demand_values = problem.get_timeseries(
                     f"{demand}.target_heat_demand", ensemble_member
                 ).values
             except KeyError:
                 continue
-            if total_demand is None:
-                total_demand = demand_values
+            if total_heat_demand is None:
+                total_heat_demand = demand_values
             else:
-                total_demand += demand_values
+                total_heat_demand += demand_values
             heat_demand_nominal[f"{demand}.Heat_demand"] = max(demand_values)
             heat_demand_nominal[f"{demand}.Heat_flow"] = max(demand_values)
 
-        # TODO: the approach of picking one peak day was introduced for a network with a tree
-        #  layout and all big sources situated at the root of the tree. It is not guaranteed
-        #  that an optimal solution is reached in different network topologies.
-        idx_max = int(np.argmax(total_demand))
-        max_day = idx_max // 24
-        nr_of_days = len(total_demand) // 24
+        total_cold_demand = None
+        cold_demand_nominal = dict()
+        for demand in cold_demands:
+                try:
+                    cold_demand_values = problem.get_timeseries(
+                        f"{demand}.target_cold_demand", ensemble_member
+                    ).values
+                except KeyError:
+                    continue
+                if total_cold_demand is None:
+                    total_cold_demand = cold_demand_values
+                else:
+                    total_cold_demand += cold_demand_values
+                cold_demand_nominal[f"{demand}.Cold_demand"] = max(cold_demand_values)
+                cold_demand_nominal[f"{demand}.Heat_flow"] = max(cold_demand_values)
+
+        idx_max_hot = int(np.argmax(total_heat_demand))
+        max_day_hot = idx_max_hot // 24
         new_date_times = list()
+        nr_of_days = len(total_heat_demand) // 24
         day_steps = problem_day_steps
 
-        problem_indx_max_peak = max_day // day_steps
-        if max_day % day_steps > 0:
-            problem_indx_max_peak += 1.0
+        problem_indx_max_peak = max_day_hot // day_steps
+        if max_day_hot % day_steps > 0:
+            problem_indx_max_peak += 1.0       
+
+        idx_max_cold = int(np.argmax(total_cold_demand))
+        max_day_cold = idx_max_cold // 24
 
+        problem_indx_max_peak_cold = max_day_cold // day_steps
+        if max_day_cold % day_steps > 0:
+            problem_indx_max_peak_cold += 1.0
```

Comment:
problem_indx_max_peak_cold doesn't seem to be used. prblem_indx_max_peak is returned at the end of the method. Check if this is also necessary for peak_cold.

Reply:
This one I am not sure about. Indeed problem_indx_max_peak is returned with some other arguments at the end of the method, but the outputs are not used in the example file. I will have to check deeper into the code to see if these output arguments are needed at all.

Reply:
I looked into this one. I don't see anywhere a need for problem_indx_max_cold outside of this function. My suggestion is to leave it as is (with it being declared and not used). And if later in the future this value is needed, add it to the return list. The problem I see with adding it to the return list now, is that it changes the number of outputs of adapt_profiles.py. This would require changing every single call to adapt_profiles.py throughout the code base. Seems like a lot of work now for a variable that is not needed right now.